react-frontend/src/tests/api.test.js

```diff
@@ -0,0 +1,292 @@
+// api.test.js
+import axios from 'axios';
+import * as api from '../api';
+
+jest.mock('axios');
+
+describe('API functions', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('register_ValidInput_ReturnsUserData', async () => {
+    // Arrange
+    const mockResponse = { data: { user_id: '123', message: 'User created successfully' } };
+    axios.post.mockResolvedValue(mockResponse);
+    const email = 'test@example.com';
+    const password = 'password123';
+    const name = 'John';
+    const surname = 'Doe';
+
+    // Act
+    const result = await api.register(email, password, name, surname);
+
+    // Assert
+    expect(axios.post).toHaveBeenCalledWith('/auth/register', { email, password, name, surname });
+    expect(result).toEqual(mockResponse.data);
+  });
+
+  test('login_ValidCredentials_ReturnsSuccessMessage', async () => {
```

Comment:
In Jest, we can use just human-readable text

Reply:
I know, but I really prefer naming convention. If it is compulsory I could change all names of the tets. By the way, I have fixed unti tests in [2ca8b01](https://github.com/YaroslavKSE/AuctionWeb/pull/3/commits/2ca8b0103fce45c94fff0ebf1ab32706383a1f05)